sites/en/intro-to-rails/getting_started.step

```diff
@@ -41,15 +41,69 @@ steps do
   end
 
   step do
-    console "rails new suggestotron"
-    message "`rails new` creates a new Rails project with the name you give."
-    message "In this case we told it to create a new project called `suggestotron`. We'll go into detail on what it created shortly."
-    message "This will print a lot of stuff to the screen and can take a while to finish."
-  end
+    message "Create the new Rails app:"
+
+    option_half "Without Docker" do
+      console "rails new suggestotron"
+      message "`rails new` creates a new Rails project with the name you give."
+      message "In this case we told it to create a new project called `suggestotron`. We'll go into detail on what it created shortly."
+      message "This will print a lot of stuff to the screen and can take a while to finish."
+      console "cd suggestotron"
+      message "`cd suggestotron` makes suggestotron our current directory."
+    end
 
-  step do
-    console "cd suggestotron"
-    message "`cd suggestotron` makes suggestotron our current directory."
+    option_half "With Docker" do
+      console "mkdir suggestotron"
+      console "cd suggestotron"
+
+      message "Copy this into the file `Gemfile`:"
+
+      source_code :ruby, <<-CONTENTS
+				source "https://rubygems.org"
+				gem "rails", "5.1.4"
+      CONTENTS
+
+      console "touch Gemfile.lock"
+
+      message "Copy this into the file `Dockerfile`:"
+
+      source_code :text, <<-CONTENTS
+				FROM ruby:2.5.0-slim
+
+				RUN apt-get update -qq && apt-get install -y build-essential libpq-dev nodejs postgresql-client libsqlite3-dev build-essential
+
+				RUN mkdir /app
+				WORKDIR /app
+				COPY Gemfile /app/Gemfile
+				COPY Gemfile.lock /app/Gemfile.lock
+				RUN bundle install
+				COPY . /app
+      CONTENTS
+
+			message "Copy this into the file `docker-compose.yml`:"
+
+			source_code :text, <<-CONTENTS
+        version: '3'
+        services:
```

Comment:
We need a db service in order for rails to work.  There's https://hub.docker.com/_/postgres/ and https://hub.docker.com/_/mysql/ depending on what database we're using.

Reply:
By default it will use sqlite3, right? The intro to rails curriculum has them using sqlite3 in development and then adding Postgres in order to deploy to heroku (which doesn’t allow sqljte3). Using Postgres without docker is pretty complicated. Using it with docker is pretty easy. But if we switch to Postgres for development for the docker students, I sort of feel like we should also switch for the non-docker students, to make the tracks continue in parallel as much as possible. But I’m not sure that’s possible given the complexity of running Postgres without docker. That’s what I was thinking by omitting that step for now. I think it will work — it should produce a database file which will be gitignored and available on the filesysrem because of the volume mount. What do you think? Happy to reconsider

Reply:
Ugh. ok.  There is a docker image for sqlite3 that we can use.  I think if we're going to use docker we should run everything within docker.
I wouldn't be opposed to having docker people just use postgresql and just have them skip the step where people have to switch from sqlite3 to postgresql to deploy to heroku.  It is more realistic to use postgresql for development, imho.